rskj-core/src/main/java/org/ethereum/db/MutableRepository.java

```diff
@@ -299,6 +299,10 @@ public synchronized Coin getBalance(RskAddress addr) {
     public synchronized Coin addBalance(RskAddress addr, Coin value) {
         AccountState account = getAccountStateOrCreateNew(addr);
 
+        if (value.equals(new Coin(BigInteger.ZERO))) {
+            return account.getBalance();
```

Comment:
What does it happen if you send 0 balance to a non-existent account? Should it be created?

Reply:
Another question: you add 0 balance to an account and then it updates its state. When it updates its state it invokes `internalPut` with the new AccountState encoded. Yet, the new AccountState should be equal to the old one. With the change in the `internalPut` method, it shouldn't be added, should it?

Reply:
Well we can avoid this line and put all of the responsibility to the `internalPut`, if addBalance(0) doesn't change anything, the internalPut method won't change anything

Reply:
> What does it happen if you send 0 balance to a non-existent account? Should it be created?

> I had the same question. I guess it doesn't matter that it is not created?

Now the `addBalance` method is not changed

Reply:
In Ethereum, currently when you add 0 to an account that has not been created, it is not created.
AFAIK, in RSK, it is.